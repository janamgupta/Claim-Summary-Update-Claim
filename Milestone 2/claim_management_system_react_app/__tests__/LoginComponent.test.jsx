/* eslint-disable no-unused-vars */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../component/LoginComponent';
import Axios from 'axios';

jest.mock('Axios');
describe('Login', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// test('should fetch users', () => {
//     const users = [{"_id":1,"username":"abc12","password":"abc12"},{"_id":2,"username":"pqr12","password":"pqr12"}];
//     const resp = {data: users};
//     Axios.get.mockResolvedValue(resp);
//     let login = new Login();  
//     login.userValidation().then(data => expect(data).toEqual(users));
//   });

