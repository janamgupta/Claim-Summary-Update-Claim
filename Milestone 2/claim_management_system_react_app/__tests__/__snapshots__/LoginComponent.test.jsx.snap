// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="logincard mx-auto card"
    style={
      Object {
        "marginBottom": "3%",
        "maxWidth": "20rem",
      }
    }
    variant="border-dark"
  >
    <div
      className="text-dark card-body"
    >
      <form
        className="loginform"
      >
        <p
          className="h4 mb-4 text-center"
        >
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-user-circle fa-w-16 "
            data-icon="user-circle"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
            Sign in
        </p>
        <div
          className="form-row"
        >
          <div
            className="form-group col-md-12"
          >
            <input
              autoComplete="off"
              className="form-control"
              name="userName"
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
            <div
              className="error-msg"
            >
              
            </div>
          </div>
        </div>
        <div
          className="form-row"
        >
          <div
            className="form-group col-md-12"
          >
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <div
              className="error-msg"
            >
              
            </div>
          </div>
        </div>
        <div
          className="text-center"
        >
          <button
            className="my-4 signin btn btn-outline-primary"
            disabled={true}
            onClick={[Function]}
            type="submit"
          >
             SignIn  
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-sign-in-alt fa-w-16 "
              data-icon="sign-in-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </div>
      </form>
      <h5
        className="text-center"
      />
    </div>
  </div>
</div>
`;
